scenario/consistency_checker.py

```diff
@@ -179,6 +182,82 @@ def check_secrets_consistency(
     return Results(errors, [])
 
 
+def check_relation_consistency(
+    *, state: "State", event: "Event", charm_spec: "_CharmSpec", **_kwargs
+) -> Results:
+    errors = []
+    nonpeer_relations_meta = list(
+        chain(
+            charm_spec.meta.get("requires", {}).items(),
+            charm_spec.meta.get("provides", {}).items(),
+        )
+    )
+    peer_relations_meta = charm_spec.meta.get("peers", {}).items()
+    all_relations_meta = list(chain(nonpeer_relations_meta, peer_relations_meta))
+
+    def _get_relations(r):
+        try:
+            return state.get_relations(r)
+        except ValueError:
+            return ()
```

Comment:
It is perfectly fine to have declared a relation endpoint in metadata.yaml, but it so happens that at this very moment the relation is not there (i.e. if you were to call `relation-list`, you'd get a null reply or an error). For example because we're in the startup sequence.

That is what this routine checks: that IF there is a relation specified in metadata.yaml, AND there is a corresponding relation provided to the State (i.e. a relation that is currently accessible via hook tools), then [relevant consistency checks]